iree/compiler/Conversion/Common/RemoveDeadMemAllocsPass.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//===-RemoveDeadMemAllocsPass.cpp - Pass to remove dead alloc-like ops ----===//
+//
+// Pass to remove operations with Allocate MemoryEffects when the allocations
+// are dead.
+//
+//===----------------------------------------------------------------------===//
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Interfaces/SideEffectInterfaces.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+struct RemoveDeadMemAllocs : RewritePattern {
```

Comment:
Can't this be `OpRewritePattern< MemoryEffectOpInterface>`?

Reply:
I think that doesnt work. ```default constructor of 'RemoveDeadMemAllocs' is implicitly deleted because base class 'OpRewritePattern<mlir::MemoryEffectOpInterface>' has no default constructor```

Reply:
got you, thanks.